contrib/rthooks/tetragon-oci-hook/cmd/hook/main.go

```diff
@@ -18,41 +18,44 @@ import (
 	"log/slog"
 	"os"
 	"os/signal"
+	"path"
 	"path/filepath"
 	"strings"
 	"syscall"
 	"time"
 
+	"github.com/alecthomas/kong"
 	"github.com/cilium/lumberjack/v2"
 	"github.com/cilium/tetragon/api/v1/tetragon"
 	"github.com/opencontainers/runc/libcontainer/cgroups/systemd"
 	"github.com/opencontainers/runtime-spec/specs-go"
-	flag "github.com/spf13/pflag"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 )
 
 var (
-	logFname                        = flag.String("log-fname", "/var/log/tetragon-oci-hook.log", "log output filename")
-	logLevelStr                     = flag.String("log-level", "info", "log level")
-	agentAddress                    = flag.String("grpc-address", "unix:///var/run/cilium/tetragon/tetragon.sock", "gRPC address for connecting to the tetragon agent")
-	grpcTimeout                     = flag.Duration("grpc-timeout", 10*time.Second, "timeout for connecting to agent via gRPC")
-	disableGrpc                     = flag.Bool("disable-grpc", false, "do not connect to gRPC address. Instead, write a message to log")
-	annotationsNamespaceDefaultKeys = []string{"io.kubernetes.pod.namespace", "io.kubernetes.cri.sandbox-namespace"}
-	annotationsNamespaceKeys        = flag.StringSlice(
-		"annotations-namespace-key",
-		annotationsNamespaceDefaultKeys,
-		"Runtime annotations  key for kubernetes namespace")
-	failCelUser = flag.String(
-		"fail-cel-expr",
-		"",
-		"CEL expression to decide whether to fail (and stop container from starting) or not.")
-	failAllowNamespaces = flag.StringSlice(
-		"fail-allow-namespaces",
-		[]string{"kube-system"},
-		"Command will not fail for specified namespaces. The namespace is determined by runtime annotation labels. Flag will be ignored if fail-cel-expr is set.")
+	binDir                          = getBinaryDir()
+	defaultLogFname                 = filepath.Join(binDir, "tetragon-oci-hook.log")
+	defaultConfFile                 = filepath.Join(binDir, "tetragon-oci-hook.json")
+	defaultAgentAddress             = "unix:///var/run/cilium/tetragon/tetragon.sock"
+	defaultAnnotationsNamespaceKeys = "io.kubernetes.pod.namespace,io.kubernetes.cri.sandbox-namespace"
+	defaultAllowNamspaces           = "kube-system"
 )
 
+var cliConf struct {
+	LogFname            string        `name:"log-fname" default:"${defLogFname}" help:"log output filename."`
```

Comment:
oh nice templating defaults, why are we not doing it everywhere :D

Reply:
Yeah, thought I try out `kong` instead of cobra+viper, and so far it seems like the better option.